cmd/snap-bootstrap/cmd_initramfs_mounts.go

```diff
@@ -405,7 +434,33 @@ func generateMountsModeRun(mst *initramfsMountsState) error {
 		return err
 	}
 
-	// 4.1. read modeenv
+	// 4.1 verify that ubuntu-data comes from where we expect it to
+
+	// since ubuntu-data has already been mounted, see if there is an encrypted
+	// device that we should try to boot from
+	diskOpts := &disks.Options{}
+	_, err = disk.FindMatchingPartitionUUID("ubuntu-data-enc")
```

Comment:
(nitpick^2) This could be written in a slightly more compact way (single line): `if _, err := ...; err == nil {` which we tend to use a lot.

Reply:
sure, sorry I am more used to putting the err := ... on it's own line but I see that you prefer it always on the same line :smile:

Reply:
this bit actually ended up going away with Samuele's suggested refactor